project-koku/koku
# [COST-3796] Migration for AWS cost category settings move

## Jira Ticket

[COST-3796](https://issues.redhat.com/browse/COST-3796)

## Description

This change will add a uuid to the aws cost category model. 

## Testing
1. Checkout main
2. make create-test-customer
3. make load-test-customer-data test_source=AWS
4. check database:
```
postgres=# \dt reporting_aws*
Did not find any relation named "reporting_aws*".
postgres=# set search_path=org1234567;
SET
postgres=# select * from reporting_awsenabledcategorykeys;
     key      | enabled 
--------------+---------
 CostCenter   | t
 Organization | t
 Charge type  | t
 OUs          | t
 env          | t
```
5. checkout this branch
6. make run-migrations
7. check database:
```
postgres=# set search_path=org1234567;
SET
postgres=# select * from reporting_awsenabledcategorykeys;
     key      | enabled |                 uuid                 
--------------+---------+--------------------------------------
 CostCenter   | t       | 090e4301-f58c-4427-9a17-8cdafe8c1bb1
 Organization | t       | 24c3460e-4054-466d-a0ba-9b3af388dbfe
 Charge type  | t       | 229f56ba-8d70-4748-b641-c3bb6cb25243
 OUs          | t       | 1d67bd29-7165-46ab-987a-ed803c2c239f
 env          | t       | 90ac1f84-adee-4c60-8b2d-55e0acadec75
```
8. Use current settings to update enabled keys:
URL: http://localhost:8000/api/cost-management/v1/settings/
Type: POST
Body:
```
{
    "api": {
        "settings": {
            "currency": "USD",
            "tag-management": {
                "enabled": [
                ]
            },
            "cost_type": "savingsplan_effective_cost",
            "aws-category-management": {
                "enabled": [
                    "aws-env"
                ]
            }
        }
    }
}
```
9. Check database:
```
postgres=# select * from reporting_awsenabledcategorykeys;
     key      | enabled |                 uuid                 
--------------+---------+--------------------------------------
 env          | t       | 90ac1f84-adee-4c60-8b2d-55e0acadec75
 CostCenter   | f       | 090e4301-f58c-4427-9a17-8cdafe8c1bb1
 Organization | f       | 24c3460e-4054-466d-a0ba-9b3af388dbfe
 Charge type  | f       | 229f56ba-8d70-4748-b641-c3bb6cb25243
 OUs          | f       | 1d67bd29-7165-46ab-987a-ed803c2c239f
```
10. Check that new keys can be added after migrationis run, with a fresh db on this branch:
```
make create-test-customer
make load-test-customer-data test_source=AWS
```
11. Check Logs:
```
koku-koku-worker-1  | [2023-06-14 14:47:16,082] INFO bde8d5a5-243a-4866-a4c7-1da6bb52ef54 Creating enabled key records: awsenabledtagkeys.
koku-koku-worker-1  | [2023-06-14 14:47:16,084] INFO bde8d5a5-243a-4866-a4c7-1da6bb52ef54 Create batch 1: batch_size 7
koku-koku-worker-1  | [2023-06-14 14:47:16,087] INFO bde8d5a5-243a-4866-a4c7-1da6bb52ef54 Creating enabled key records: awsenabledcategorykeys.
koku-koku-worker-1  | [2023-06-14 14:47:16,088] INFO bde8d5a5-243a-4866-a4c7-1da6bb52ef54 Create batch 1: batch_size 5
koku-koku-worker-1  | [2023-06-14 14:47:16,128] INFO bde8d5a5-243a-4866-a4c7-1da6bb52ef54 Checking for schema
koku-koku-worker-1  | [2023-06-14 14:47:16,178] INFO bde8d5a5-243a-4866-a4c7-1da6bb52ef54 Checking for table
koku-koku-worke
```
12. Check category keys:
```
postgres=# select * from reporting_awsenabledcategorykeys;
     key      | enabled |                 uuid                 
--------------+---------+--------------------------------------
 env          | t       | 8f7e988d-488e-43df-9b1f-ad8deca3983d
 Charge type  | t       | 485a9bfe-c787-43bf-8417-def83eb15f0d
 Organization | t       | 7f8c5688-dc92-4b12-8d4f-c882faa65e67
 OUs          | t       | b6e8303b-6d58-49cc-9962-ab419e6b13c9
 CostCenter   | t       | 30bf5072-d2dd-4d07-9a4d-9d02a4c2048d
```


## Notes

...
